pkg/utils/host/host.go

```diff
@@ -65,11 +70,115 @@ func init() {
 	}
 	HostProcFs = filepath.Join(HostRoot, "/proc")
 
+	err := workarounds()
+	if err != nil {
+		panic(err)
+	}
+
 	// Initialize IsHost*Ns
 	IsHostPidNs = isHostNamespace("pid")
 	IsHostNetNs = isHostNamespace("net")
 }
 
+func workarounds() error {
+	// No memory limit for eBPF maps
+	if err := rlimit.RemoveMemlock(); err != nil {
+		return err
+	}
+
+	// Some environments (e.g. minikube) runs with a read-only /sys without bpf
+	// https://github.com/kubernetes/minikube/blob/99a0c91459f17ad8c83c80fc37a9ded41e34370c/deploy/kicbase/entrypoint#L76-L81
+	// Docker Desktop with WSL2 also has filesystems unmounted.
+	// Ensure filesystems are mounted correctly.
+	fs := []struct {
+		name  string
+		path  string
+		magic int64
+	}{
+		{
+			"bpf",
+			"/sys/fs/bpf",
+			unix.BPF_FS_MAGIC,
+		},
+		{
+			"debugfs",
+			"/sys/kernel/debug",
+			unix.DEBUGFS_MAGIC,
+		},
+		{
+			"tracefs",
+			"/sys/kernel/tracing",
+			unix.TRACEFS_MAGIC,
+		},
+	}
+	for _, f := range fs {
+		var statfs unix.Statfs_t
+		err := unix.Statfs(f.path, &statfs)
+		if err != nil {
+			return fmt.Errorf("statfs %s: %w", f.path, err)
+		}
+		if statfs.Type == f.magic {
+			log.Debugf("%s already mounted", f.name)
+		} else {
+			err := unix.Mount("none", f.path, f.name, 0, "")
+			if err != nil {
+				return fmt.Errorf("mounting %s: %w", f.path, err)
+			}
+			log.Debugf("%s mounted (%s)", f.name, f.path)
+		}
+	}
+
+	// Docker Desktop with WSL2 sets up host volumes with weird pidns.
+	if HostRoot != "/" {
```

Comment:
I didn't find any environment variable that distinguish Windows or WSL2. GOHOSTOS would be Linux because `ig` is compiled for Linux and is running on the Linux kernel.
```
$ uname -r
5.10.102.1-microsoft-standard-WSL2
```

But then I thought it is better to test the actual problem rather that test the OS. The reason we need this workaround is because the procfs `/host/proc` is mounted in a pid namespace which is neither the pid namespace of the ig container or a parent pid namespace (therefore it is not the host pidns too). This seems to be a recent change in Docker Desktop for WSL2. So I am testing that below with `readlink("/host/proc/self")`.

If Docker Desktop for WSL2 changes its implementation to no longer do that, then the workaround will not be used anymore.